amoro-mixed-format/amoro-mixed-format-flink/amoro-mixed-format-flink-common/src/main/java/org/apache/amoro/flink/catalog/FlinkUnifiedCatalog.java

```diff
@@ -221,8 +226,25 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     TableIdentifier tableIdentifier =
         TableIdentifier.of(
             unifiedCatalog.name(), tablePath.getDatabaseName(), tablePath.getObjectName());
+    if (!configuration.contains(TABLE_FORMAT)) {
+      // if user doesn't specify the table format, we try to load the table and get the table
+      // format.
+      try {
+        AmoroTable amoroTable =
+            unifiedCatalog.loadTable(tableIdentifier.getDatabase(), tableIdentifier.getTableName());
+        format = amoroTable.format();
+      } catch (Throwable t) {
+        LOG.warn(
+            "We can't load table {}.{} through unifiedCatalog, use default format {}",
+            tablePath.getDatabaseName(),
+            tablePath.getObjectName(),
+            format.name());
+      }
+    }
+    final TableFormat catalogFormat = format;
```

Comment:
This method is used to create a table. How to load the table before creating it?

Reply:
> This method is used to create a table. How to load the table before creating it?
```
  public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfExists)
```
In the Flink SQL scenario, when users are writing SQL, this situation arises when they use "create table if not exists" to define a table that already exists. The parameter "ignoreIfExists" is used to ignore the existence of the table.